### PR TITLE
Run IWYU (with header reordering)

### PR DESCRIPTION
**Pull request by jbytheway**
#### Summary
SUMMARY: None

#### Purpose of change
Keep header status up to date.

Verify that IWYU is still usable on CDDA.

#### Describe the solution
Run IWYU & astyle.

This time I ran IWYU with `--reorder` so the headers are sorted alphabetically.

#### Testing
Superficial glancing at the results.

#### Additional context
Had to fix various IWYU bugs for this to work.  Not all merged yet, so not trivial for others to reproduce this yet.

### CODE REVIEW

File: tools/clang-tidy-plugin/TranslatorCommentsCheck.h [hunk 1]
```diff
@@ -1,6 +1,10 @@
 #ifndef CATA_TOOLS_CLANG_TIDY_TRANSLATORCOMMENTSCHECK_H
 #define CATA_TOOLS_CLANG_TIDY_TRANSLATORCOMMENTSCHECK_H
 
+#include <clang/ASTMatchers/ASTMatchFinder.h>
+#include <clang/Basic/SourceLocation.h>
+#include <llvm/ADT/StringRef.h>
+#include <memory>
 #include <set>
 
 #include "ClangTidy.h"
```
Comment (Qrox) [inline tools/clang-tidy-plugin/TranslatorCommentsCheck.h:10]:
```suggestion
#include <ClangTidy.h>
```
for consistency.

File: tools/clang-tidy-plugin/TranslatorCommentsCheck.cpp [hunk 2]
```diff
@@ -1,9 +1,35 @@
...
+#include <clang/Lex/PPCallbacks.h>
+#include <clang/Lex/Preprocessor.h>
+#include <clang/Lex/Token.h>
+#include <llvm/ADT/STLExtras.h>
+#include <llvm/Support/Regex.h>
+#include <iterator>
+#include <map>
+#include <utility>
+
+#include "clang/Basic/TokenKinds.h"
```
Comment (Qrox) [inline tools/clang-tidy-plugin/TranslatorCommentsCheck.cpp:26]:
```suggestion
#include <clang/Basic/TokenKinds.h>
```

File: tools/clang-tidy-plugin/NoStaticGettextCheck.cpp [hunk 3]
```diff
@@ -1,7 +1,10 @@
 #include "NoStaticGettextCheck.h"
 
+#include <clang/AST/Expr.h>
+#include <clang/ASTMatchers/ASTMatchers.h>
+#include <clang/ASTMatchers/ASTMatchersInternal.h>
+
 #include "clang/ASTMatchers/ASTMatchFinder.h"
```
Comment (Qrox) [inline tools/clang-tidy-plugin/NoStaticGettextCheck.cpp:7]:
```suggestion
#include <clang/ASTMatchers/ASTMatchFinder.h>
```

**Comment by jbytheway** [comment]
I forgot that `clang-tidy` is enforcing using the C++ versions of the C std lib headers, so this will require some further code changes.  Closing for now.  I'll also include @Qrox's suggestions for the next version.

File: src/active_item_cache.h [hunk 4]
```diff
@@ -2,12 +2,14 @@
 #ifndef ACTIVE_ITEM_CACHE_H
 #define ACTIVE_ITEM_CACHE_H
 
+#include <iosfwd>
 #include <list>
+#include <system_error>
```
[the remark this answers is not in the document]
Comment (jbytheway) [reply src/active_item_cache.h:7]: This was for `std::hash`, which I guess happens yo be included by `system_error`.  I've fixed that in the mapping file so `std::hash` now causes `functional` to be included instead.

**Pull request closed without merging**